gtsam/linear/GaussianConditional.cpp

```diff
@@ -299,14 +299,22 @@ double GaussianConditional::logDeterminant() const {
           "GaussianConditional::sample can only be called on single variable "
           "conditionals");
     }
+
+    // Noise model to sample from.
+    noiseModel::Diagonal::shared_ptr _model;
     if (!model_) {
-      throw std::invalid_argument(
-          "GaussianConditional::sample can only be called if a diagonal noise "
-          "model was specified at construction.");
+      // Initialize model_ to Unit noiseModel if not initialized.
+      // Unit noise model specifies sigmas as 1, since
+      // the noise information should already be in information matrix A.
+      // Dim should be number of rows since
+      // noise model dimension should match (Ax - b)
+      _model = noiseModel::Unit::Create(rows());
```

Comment:
No, sorry, you can’t do that. This method is and should stay const. Just pass the a vector of one’s in sampleDiagonal below.

Reply:
It is const. I am assigning a new model `_model` (not `model_`) based on the conditions. This is functional since there is no side-effect. 🙂

Reply:
Okay made it a lot cleaner. Hopefully it's also good for you.